pkg/scrape/target.go

```diff
@@ -108,22 +108,20 @@ func (t *Target) Params() url.Values {
 
 // Labels returns a copy of the set of all public labels of the target.
 func (t *Target) Labels() labels.Labels {
-	lset := make(labels.Labels, 0, len(t.labels))
-	for _, l := range t.labels {
+	b := labels.NewScratchBuilder(t.labels.Len())
```

Comment:
For reference:
https://github.com/prometheus/prometheus/blob/2d588725c12e46e4e71da3899ec05bb933151021/model/labels/labels.go#L576